platforms/core-configuration/declarative-dsl-core/src/main/resources/org/gradle/internal/declarativedsl/parsing/grammar.bnf

```diff
@@ -43,27 +48,29 @@
 
 root ::= statements
 
-statement ::= (assignment | block | functionChain)
+statement ::= (assignment | block | expression)
 statements ::= (statement SEMI*)*
 
-assignment ::= directlyAssignableExpression (EQ|PLUSEQ) assignmentValue {pin=2}
 directlyAssignableExpression ::= simpleIdentifier navigationSuffix?
-assignmentValue ::= (simpleIdentifier (navigationSuffix* valueArguments)?) | literalOrIdentifier
+
+assignment ::= directlyAssignableExpression (EQ|PLUSEQ) expression {pin=2}
 
 block ::= simpleIdentifier valueArguments? annotatedLambda
 annotatedLambda ::= LB statements RB {pin=1}
 
-functionChain ::= simpleIdentifier postfixUnarySuffixes
-postfixUnarySuffixes ::= (navigationSuffix+ valueArguments) | (valueArguments (navigationSuffix valueArguments)*)
-
 valueArguments ::= LP (valueArgument (COMA valueArgument)* COMA?)? RP {pin=2}
-valueArgument ::= (simpleIdentifier EQ)? (functionChain | literalOrIdentifier)
+valueArgument ::= (simpleIdentifier EQ)? expression
+
+expression ::= infixFunctionCall
+infixFunctionCall ::= rangeExpression ('to' rangeExpression)?
+rangeExpression ::= simpleIdentifier postfixUnarySuffixes | primaryExpression
```

Comment:
💭 Do I get it right that `rangeExpression` is used as the name here to make it look closer to the Kotlin grammar? We don't have ranges in our language, so at first sight, this name looked surprising to me.

Reply:
These names were kept to make it more obvious how trimming down the origin Kotlin grammar leads to what we end up with. But I see how this might create more confusion than benefit. Will reconsider.

Reply:
Won't earn me an award for naming excellence, but this is what I could come up with:

```
expression ::= simpleExpression ('to' simpleExpression)?
simpleExpression ::= simpleIdentifier postfixUnarySuffixes | primaryExpression
```